test_suite/ui_tests.rs

```diff
@@ -12,3 +12,18 @@ fn ui() {
         }
     }
 }
+
+#[cfg(windows)]
+#[test]
+fn ui() {
+    use glob::glob;
+    let t = trybuild::TestCases::new();
+    for test in glob("ui/*.rs").unwrap() {
+        let path = test.unwrap();
+        if path == std::path::Path::new("ui/must_use_warning.rs") {
+            t.pass(path)
+        } else {
+            t.compile_fail(path)
+        }
+    }
+}
```

Comment:
I don't think there's any benefit to the implementation that uses Unix-specific features. Why not merge these?